src/foam/u2/DAOChoicesListView.js

```diff
@@ -28,51 +26,48 @@ foam.CLASS({
   properties: [
     'of',
     {
-      name: 'sections',
+      name: 'choices',
       value: []
     },
-    {
-      name: 'search',
-      value: true
-    },
-    {
-      name: 'choosePlaceholder',
-      value: 'Choose Collection...'
-    }
   ],
 
   methods: [
     function init() {
-      this.SUPER();
+      var flowChoices = [];
+      var daoChoices = [];
 
-      var flowChildrenDAOs = foam.dao.MDAO.create({of: foam.core.boot.CSpec});
-      var flowSection = this.RichChoiceViewSection.create({
-        heading: 'Flow DAOs',
-        dao: flowChildrenDAOs
-      });
-      var daoSection = this.RichChoiceViewSection.create({
-        heading: 'DAOs',
-        dao: this.cSpecDAO.where(
-          this.AND(
-            this.ENDS_WITH(foam.core.boot.CSpec.ID, 'DAO'),
-            this.EQ(foam.core.boot.CSpec.SERVE, true)
-          )
+      this.flowChildren.forEach( child => {
+          child.value?.cls_ && child.value.cls_.getAxiomsByClass(foam.dao.DAOProperty).forEach( prop => flowChoices.push(child.flowName + '.' + prop.name) )
+      })
+      
+      // Question to ask/look into: 
+      // Why is setting of this.choices here(LN:47) important? Because if 
+      // - I remove Line::(`this.choices = [...flowChoices]`) then things stop working on first match command 
+      // If I filter flowChoices at Line:39 right before pushing, then first match command nothing renders
+      this.choices = [ ...flowChoices ]
+      
+      var allDAOs = this.cSpecDAO.where(
```

Comment:
You can do this.CSpecDAO.where(foam.core.boot.CSpec.SERVED_DAOS), since Served_DAOS is a "canned query" in the model.